core/runtime.rs

```diff
@@ -2412,9 +2508,13 @@ pub fn queue_fast_async_op(
     // atleast 1 Rc is held by the JsRuntime.
     None => unreachable!(),
   };
+  let realm_idx = ctx.realm_idx;
 
   let mut state = runtime_state.borrow_mut();
-  state.pending_ops.push(OpCall::lazy(op));
+  state.pending_ops.push(OpCall::lazy(async move {
+    let (promise_id, op_id, resp) = op.await;
+    (realm_idx, promise_id, op_id, resp)
+  }));
```

Comment:
pass `op: impl Future<Output = (RealmId, PromiseId, OpId, OpResult)> + 'static,` from the caller to avoid this extra future creation here. It's done in deno_ops

Reply:
Should I do that as well for `queue_async_op`, even if the realm index will get discarded in the eager case?

Reply:
yes

Reply:
Done